### ISSUE TITLE
Native Pages Functionality

### ISSUE THREAD

**Issue by ChristopherBiscardi** [ChristopherBiscardi/gatsby-mdx#4]
# Desired Behavior

A user enables `gatsby-mdx` in their `gatsby-config.js`. They can then use `.mdx` files in `src/pages/*` immediately to create new pages on their site.

```js
// gatsby-config.js
module.exports = {
  siteMetadata: {
    title: `my great site`
  },
  plugins: [
    `gatsby-plugin-emotion`,
    `gatsby-plugin-react-helmet`,
    { resolve: require.resolve(`gatsby-mdx`), options: {} },
  ]
};
```

```mdx
import Box from 'superbox/emotion';
import { graphql } from 'gatsby';

export default ({ children, ...props }) => <Box>{children}</Box>

export const pageQuery = graphql`
  allAuthors { edges { node { name } } }
`

# Title

some content
```

**Comment by ChristopherBiscardi** [comment]
the [page-creator plugin](https://github.com/gatsbyjs/gatsby/blob/4b144da912f33912ac80b22741886191a6a91051/packages/gatsby-plugin-page-creator/src/gatsby-node.js#L26) is enabled by default on all gatsby sites. Would be nice to send configuration to this somehow rather than re-implement the functionality.

**Comment by ChristopherBiscardi** [comment]
Works fine with the webpack loader for most things. https://gatsby-mdx-kitchen-sink.netlify.com/

Still need to figure out pageQuery in .mdx pages: #11

**Comment by avigoldman** [comment]
Still also have to solve the layout props issue: #9

**Comment by ChristopherBiscardi** [comment]
looks like the layout props PR got cleaned up this morning and uses obj/rest spread now. very awesome.

**Comment by avigoldman** [comment]
https://github.com/mdx-js/mdx/pull/189 was merged!

**Comment by ChristopherBiscardi** [comment]
yeah, it's dope. I've been using StaticQuery on my layout components right now as a bypass until the next mdx release with the props.

**Comment by avigoldman** [comment]
Looks like https://github.com/mdx-js/mdx/tree/v0.15.0-1 has the layout prop fix. We should bump the peer dependency version for `@mdx-js/mdx` and add an example around using the props Gatsby passes to pages.

I don't think there is anything else needed for MDX to be used as first-class pages!

**Comment by ChristopherBiscardi** [comment]
yup, doing that along with looking into #25

**Comment by avigoldman** [comment]
#25 is an unrelated issue – the loader is working but the code generated in the `code` GraphQL field isn't using the plugins.

**Comment by ChristopherBiscardi** [comment]
yup. Just saying I was looking into both at the same time because I created an additional examples site for the custom remark-toc application 😅

**Comment by avigoldman** [comment]
My bad 😬 Totally didn't follow 😆
